### PR TITLE
Improve tests for ProductsContainer and refactor BestSellers and ProductsPage tests

### LINKED ISSUE #286: Refactor tests for products
There are tests for components like `ProductsPage`, `BestSellers` which uses `ProductsContainer` inside. Would be good to refactor those tests and mock `ProductsContainer`. `ProductsContainer` is already tested, in its test file, so no need to duplicate tests in other files that should test its parent component logic. Also, because of current test duplication, changes in tests need to be done in few files when we make changes to `ProductsContainer` that require test changing.

### PR DESCRIPTION
**Pull request by OliaHrytsak**
image
image
image

image
image

### CODE REVIEW

File: code/src/containers/products-container/ProductsContainer.test.tsx [hunk 1]
```diff
@@ -1,68 +1,20 @@
-import { screen } from "@testing-library/react";
+import { fireEvent, screen, waitFor } from "@testing-library/react";
```
Comment (D1nviz) [inline code/src/containers/products-container/ProductsContainer.test.tsx]: whaitFor is unused

File: code/src/containers/products-container/ProductsContainer.test.tsx [hunk 2]
```diff
@@ -87,21 +40,53 @@ jest.mock("@/store/api/cartApi", () => ({
...
+  ...jest.requireActual("@/context/drawer/DrawerContext"),
+  useDrawerContext: jest.fn(() => ({ openDrawer: mockOpenDrawer }))
+}));
+
+jest.mock("@/hooks/use-get-cart/useGetCart", () => ({
+  __esModule: true,
+  default: jest.fn()
+}));
+
+const mockCartItem = (items: any) => {
```
Comment (D1nviz) [inline code/src/containers/products-container/ProductsContainer.test.tsx]: Don't use `any`

File: code/src/containers/products-container/ProductsContainer.test.tsx (hunk 1 above)
Comment (OliaHrytsak) [reply code/src/containers/products-container/ProductsContainer.test.tsx]: Fixed!

File: code/src/containers/products-container/ProductsContainer.test.tsx (hunk 2 above)
Comment (OliaHrytsak) [reply code/src/containers/products-container/ProductsContainer.test.tsx]: Fixed!

**Review by heshvab — APPROVED** [review]

**Comment by heshvab** [comment]
Great work!

**Review by D1nviz — APPROVED** [review]

File: code/src/pages/products/ProductsPage.test.tsx [hunk 3]
```diff
@@ -3,34 +3,14 @@ import { fireEvent, screen } from "@testing-library/react";
...
-    getCartItems: {
-      matchFulfilled: jest.fn()
-    }
-  }
-}));
-
-jest.mock("@/store/api/productsApi", () => ({
-  useGetProductsQuery: jest.fn()
-}));
+type ProductsContainerType = {
```
Comment (dmtrth25) [inline code/src/pages/products/ProductsPage.test.tsx]: repeated type here `ProductsContainerType` the same fields uses for `BestSellers` tests.
I suggest to create some common type and reuse it

**Review by mhevyk — APPROVED** [review]

**Review by mhevyk — APPROVED** [review]
Great work!

File: code/src/pages/products/ProductsPage.test.tsx (hunk 3 above)
Comment (OliaHrytsak) [reply code/src/pages/products/ProductsPage.test.tsx]: Thank for this suggestions! I've desided to re-use existing type

**Review by dmtrth25 — APPROVED** [review]

**Pull request merged**